Add schema validation and configuration for conda-forge config yaml (#1756)
conda-forge/conda-smithy: news/1756-schema.rst

**Added:**

> * New conda-forge pydantic schema for ``conda-forge.yaml``, includes a generate default forge yaml and json schema dynamically from the pydantic model

Reviewer: ```suggestion
* New JSON schema for ``conda-forge.yaml``. A Pydantic model is used to dynamically generate both a YAML document with the default values and the JSON schema itself.
```

> * Included ``jsonschema``and ``pydantic`` as dependencies into the ``environment.yml``

Reviewer: ```suggestion
* Included ``jsonschema`` and ``pydantic`` as dependencies into the ``environment.yml``. (#1756)
```

**Changed:**

> * Included extra ``jsonschema`` validation for forge yaml, under ``configure_feedstock``

Reviewer: ```suggestion
* Included extra ``jsonschema`` validation for conda-forge.yaml, under ``configure_feedstock``. (#1756)
```

> * Moved legacy checks of old_file and providers into a new  auxiliary ``_legacy_compatibility_checks`` function

Reviewer: ```suggestion
* Moved legacy checks of old_file and providers into a new auxiliary ``_legacy_compatibility_checks`` function. (#1756)
```
